sdk/core/Azure.Core/tests/HttpMessageSanitizerTests.cs

```diff
@@ -46,5 +47,44 @@ public void QueryIsSanitizedAppendQuery()
 
             Assert.AreEqual("http://localhost/?a=*", sanitizer.SanitizeUrl(uriBuilder.ToString()));
         }
+
+        [Test]
+        public void ApiVersionIsNotSanitizedByDefault()
+        {
+            HttpMessageSanitizer sanitizer = ClientDiagnostics.CreateMessageSanitizer(ClientOptions.Default.Diagnostics);
```

Comment:
new HttpMessageSanitizer?

Reply:
You mean:
```cs
new HttpMessageSanitizer(options.Diagnostics.LoggedQueryParameters.ToArray(), options.Diagnostics.LoggedHeaderNames.ToArray());
```
Can you help me understand why this would be better?